packages/types/src/Text.ts

```diff
@@ -2,62 +2,56 @@
 // This software may be modified and distributed under the terms
 // of the ISC license. See the LICENSE file for details.
 
+import u8aConcat from '@polkadot/util/u8a/concat';
 import u8aFromUtf8 from '@polkadot/util/u8a/fromUtf8';
+import u8aToHex from '@polkadot/util/u8a/toHex';
 import u8aToUtf8 from '@polkadot/util/u8a/toUtf8';
-import u8aConcat from '@polkadot/util/u8a/concat';
 
-import Base from './codec/Base';
 import Compact from './codec/Compact';
+import { Codec } from './types';
 
 // This is a string wrapper, along with the length. It is used both for strings as well
 // as stuff like documentation.
 //
 // TODO
-//   - Strings should probably be trimmed (docs do come through with extra padding)
 //   - Potentially we want a "TypeString" extension to this. Basically something that
 //     wraps the `Balance`, `T::AccountId`, etc. The reasoning - with a "TypeString"
 //     we can nicely strip types down like "T::AcountId" -> "AccountId"
-export default class Text extends Base<string> {
+export default class Text extends String implements Codec<Text> {
   constructor (value: Text | string = '') {
     super(
-      value instanceof Text
-        ? value.raw
-        : value.trim()
+      value.trim()
     );
   }
 
-  get length (): number {
-    return this.raw.length;
-  }
-
   byteLength (): number {
     return this.length + Compact.encode(this.length).length;
   }
 
+  fromJSON (input: any): Text {
+    // FIXME Same as below
+    return new Text(`${input || ''}`);
+  }
+
   fromU8a (input: Uint8Array): Text {
     const [offset, length] = Compact.decode(input);
-
-    this.raw = u8aToUtf8(input.subarray(offset, offset + length.toNumber()));
-
-    return this;
+    // FIXME this returns a new Object unfortunately, can't "replace" current value
+    // Two solutions:
+    // - either use static
+    // - or completely remove from*, and force to use constructor
+    return new Text(u8aToUtf8(input.subarray(offset, offset + length.toNumber())));
```

Comment:
Yeap, the other PR needs to go first. (This will just create too much messy-ness)